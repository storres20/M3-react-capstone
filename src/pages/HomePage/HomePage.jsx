import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { useEffect } from 'react';
import styles from './HomePage.module.css';
import ScrollButton from '../../components/ScrollButton/ScrollButton';
import { updateMovie } from '../../redux/movies/movies';
import SearchCat from '../../components/SearchCat/SearchCat';
import { getCategory } from '../../redux/movies/categories';

export default function HomePage() {
  /* Load Redux State */
  const categories = useSelector((state) => state.category);
  // *********************************

  /* Setup Redux dispatch */
  const dispatch = useDispatch();

  /* Clean - Redux movies store */
  useEffect(() => {
    dispatch(updateMovie([]));
    dispatch(getCategory());
  }, [dispatch]);

  return (
    <div>
      <h2 className={styles.title}>Categories</h2>
      {/* Filter Search Category */}
      <SearchCat />
      <ul className={styles.moviesGrid} data-testid="moviesGrid">
        {categories.map((movie) => (
          <Link key={movie.id} to={`/detailPage/${movie.id}`}>
            <li className={styles.movieCard}>
              <div>{movie.name}</div>
            </li>
          </Link>
        ))}
      </ul>

      {/* Scroll button for going back to TOP page */}
      <ScrollButton />
    </div>
  );
}
