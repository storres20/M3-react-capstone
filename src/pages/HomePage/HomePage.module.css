.title {
  text-align: center;
  margin-top: 30px;
  margin-bottom: 0;
  font-size: 3rem;
  font-weight: bold;
}

.moviesGrid {
  display: grid;
  grid-template-columns: repeat(auto-fill, 230px);
  gap: 40px;
  padding: 40px;
  justify-content: center;
}

@media (max-width: 560px) {
  .moviesGrid {
    grid-template-columns: 100%;
  }
}

/* ************ */

.movieCard {
  list-style: none;
  font-size: 1.5rem;
  text-align: center;
  border-radius: 50px;
  height: 50px;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  background-color: #e6206f;
  box-shadow: rgba(0, 0, 0, 0.35) 0 5px 15px;
}

.movieCard:hover {
  opacity: 0.8;
  background-color: #ca3973;
}

.movieImage {
  border-radius: 10px;
  background-color: azure;
}

@media (max-width: 560px) {
  .movieImage {
    width: 90%;
    height: auto;
  }
}
