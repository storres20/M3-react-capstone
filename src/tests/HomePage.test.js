import React from 'react';
import renderer from 'react-test-renderer';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import HomePage from '../pages/HomePage/HomePage';
import categoriesReducer from '../redux/movies/categories';

// configureStore
const createTestStore = () => {
  const store = configureStore({
    reducer: {
      category: categoriesReducer,
    },
  });

  return store;
};
export default createTestStore;

// Test
describe('My Connected React-Redux Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = createTestStore();

    // Create a redux store
    component = renderer.create(
      <Provider store={store}>
        <HomePage />
      </Provider>,
    ).toJSON();
  });

  test('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
