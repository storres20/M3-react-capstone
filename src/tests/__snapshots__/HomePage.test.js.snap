// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component renders correctly 1`] = `
<div>
  <h2
    className="title"
  >
    Categories
  </h2>
  <ul
    className="moviesGrid"
  />
  <div
    className="button"
  />
</div>
`;
